src/core/version_neg.c

```diff
@@ -171,217 +171,94 @@ QuicVersionNegotiationExtGenerateCompatibleVersionsList(
 }
 
 QUIC_STATUS
-QuicVersionNegotiationExtParseClientVerNegInfo(
+QuicVersionNegotiationExtParseVersionInfo(
     _In_ QUIC_CONNECTION* Connection,
     _In_reads_bytes_(BufferLength)
         const uint8_t* const Buffer,
     _In_ uint16_t BufferLength,
-    _Out_ QUIC_CLIENT_VER_NEG_INFO* ClientVNI
+    _In_ BOOLEAN IsServer,
+    _Out_ QUIC_VERSION_INFORMATION_V1* VersionInfo
     )
 {
+    const char* const Source = (IsServer ? "Server" : "Client");
```

Comment:
Since you're logging the connection pointer with all the events already, I am actually leaning towards saying this `Source` is actually unnecessary.

Reply:
I can understand that perspective, but this makes the logs easier to read when client and server are on the same machine (testing scenarios). In production scenarios, only one side would be in the logs, probably.

Reply:
It's not that important. I'll remove it.